staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -147,3 +147,35 @@ func ComputeHistogramDelta(before, after model.Samples, label model.LabelName) {
 func makeKey(a, b model.LabelValue) string {
 	return string(a) + "___" + string(b)
 }
+
+// GetMetricValuesForLabel returns value of metric for a given dimension
+func GetMetricValuesForLabel(ms Metrics, metricName, label string) map[string]int64 {
+	result := make(map[string]int64)
+	for method, samples := range ms {
+		if method != metricName {
+			continue
+		}
+		for _, sample := range samples {
+			count := int64(sample.Value)
+			dimensionName := string(sample.Metric[model.LabelName(label)])
+			result[dimensionName] = count
+		}
+	}
+	return result
+}
+
+// HasValidMetrics verifies if every sample of metric has all expected labels
+func HasValidMetrics(metrics Metrics, metricName string, expectedLabels ...string) bool {
```

Comment:
looks like we're losing some error/log fidelity here. Is there any way we could make the return type `bool, err` instead or something to get that information back?